### ISSUE TITLE
Maximum-function-nesting-level-reached error while resolving auth guard in PHPUnit tests

### ISSUE THREAD

**Issue by bogdanghervan** [laravel/lumen-framework#349]
I have a one-week old Lumen v5.2.1 installation and I'm getting a "maximum function nesting level reached" fatal error:

```
PHP Fatal error:  Maximum function nesting level of '256' reached, aborting! in /var/www/api/src/vendor/illuminate/container/Container.php on line 700
```

(Full stack trace [here](https://github.com/laravel/lumen-framework/files/115923/maximum-function-nesting-level-reached.txt))

[Other people](https://laracasts.com/discuss/channels/lumen/error-while-testing-routes-using-auth-middleware) have also experienced this issue.

This happens only with requests dispatched from a PHPUnit test. It looks as though the authentication guard doesn't get registered properly by `AuthServiceProvider` and when the default `Authenticate` middleware attempts to retrieve it, `AuthManager` fails to resolve it, entering an infinite loop of recursive calls (`__call` gets invoked which in turn tries to resolve the guard and so on).

I found out that changing the default `AuthServiceProvider` like this works:

``` PHP
$this->app['auth']->viaRequest('api', function ($request) {
    // Return User or null...
});
```

However, [the code](https://github.com/laravel/lumen/blob/v5.2.1/app/Providers/AuthServiceProvider.php) in the v5.2.1 release lags behind:

``` PHP
Auth::viaRequest('api', function ($request) {
    // [...]
});
```

However, I still can't resolve the authenticated user from the `Illuminate\Http\Request` instance inside my route (even after it successfully passes through the `Authenticate` middleware). The only way it works is using the facade like this: `Auth::user()`.

Any thoughts?

**Comment by bogdanghervan** [comment]
Meanwhile, I see the service provider code [has been fixed](https://github.com/laravel/lumen/commit/14600a77c81a8f7656c630584c818838dc240e9f) on master by @amosmos for different reasons.

**Comment by yuloh** [comment]
It sounds like this was caused by the issue #406 fixes.  Since the service provider is fixed I think this issue could be closed.

> However, I still can't resolve the authenticated user from the Illuminate\Http\Request instance inside my route (even after it successfully passes through the Authenticate middleware). The only way it works is using the facade like this: Auth::user().

Is this when testing with phpunit or making actual HTTP requests?  There is a [passing test](https://github.com/laravel/lumen-framework/blob/5.2/tests/FullApplicationTest.php#L474) showing this working.  That fix was added in 5.2.6 so everything should be good now.

@bogdanghervan Can you make sure everything works on 5.2.6 and close this issue if it's good?

**Comment by amosmos** [comment]
I don't remember why I requested that Facade change but I'm glad it solved the issue :-)
